chain/sync.go

```diff
@@ -739,18 +758,13 @@ func (s *Syncer) Run(ctx context.Context) (err error) {
 
 	log.Infof("Blockchain sync completed, wallet ready for general usage.")
 
-	// Wait for notifications to finish before returning
-	defer func() {
-		<-s.notifier.closed
-	}()
-
 	g.Go(func() error {
 		select {
 		case <-ctx.Done():
-			client.Close()
+			walletCtxCancel()
```

Comment:
I guess doing `walletCtxCancel` here is also unnecessary (since it will be canceled with `defer` from above)